bin/src/learner/raytune_learner.py

```diff
@@ -0,0 +1,243 @@
+            raise ValueError(f"Invalid run_config: {self.config['run_config']['name']}, check PyTorch for documentation on how a run_config should be defined")
+
+        # substite in the checkpoint_config everything that is a bool in string with the actual bool
+        self.tuner = None
+        self.best_config = None
+        self.results = None
+        self.trainer = None
+
+    def _prep_tuner(self):
+        """
+        Prepare the tuner with the configs.
+        """
+        self.tuner = tune.Tuner(TuneModel,
+                            tune_config= self.tune_config,
+                            param_space=self.config,
+                            run_config=self.run_config,
+                        )
+        self.tuner.tuning = True
+
+    def tune(self, overwrite=False):
+        """
+        Run the tuning process.
+        """
+        if overwrite:
+            self.tuner = None
+            self.best_config = None
+            
+        if self.tuner is None:
+            self._prep_tuner()
+            self.results = self.tuner.fit()
+            best_config = os.path.join(self.results.get_best_result().path, "params.json")
+            try: 
+                assert os.path.exists(best_config)
+            except AssertionError:
+                raise ValueError("Best config file not found.")
+            with open(best_config, "r") as f:
+                self.best_config = eval(f.read())
+        else:
+            raise ValueError("Tuner already exists - if you want to overwrite it, please set overwrite=True.")     
+
+    def store_best_config(self, path):
+        """
+        Store the best config in a file.
+        """
+        with open(path, "w") as f:
+            f.write(str(self.best_config))            
+    
+    def train(self, config = None): 
+        """
+        Train the model with the config.
+        """
+        if config is None:
+            config = self.best_config
+        try: 
+            assert config is not None
+        except AssertionError:
+            raise ValueError("No config provided - please provide a config to train the model with or tune first by calling the tune() method.")
+        
+        self.trainer = TuneModel(config=config)
+        self.trainer.tuning = False
+        for i in range(config["epochs"]):
+            self.trainer.step()
+
+
+class TuneModel(Trainable):
+
+    def setup(self, config: dict):
+        """
+        Get the model, loss function(s), optimizer, train and test data from the config.
+        """
+
+        # Load model from string path
+        module_name, class_name = config["model"].rsplit('.', 1)
+        module = importlib.import_module(module_name)
+        model = getattr(module, class_name)
+        self.model = model(**config["model_params"])
+
+        # Add data path
+        self.data_path = config["data_path"]
+
+        # Add experiment
+        module_name, class_name = config["experiment"].rsplit('.', 1)
```

Comment:
why is overwriting the line 131?

Reply:
or better why is importing from the experiment keyword as well?

Reply:
because it is imported as a string first and then we need to import the actual class